From c361c3549e8e553976bf8250f49c0cf0ccc53b4c Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Thu, 27 Oct 2011 16:29:15 -0700
Subject: [PATCH 1/2] libparted: copy flags when duplicating GPT partitions

 * libparted/labels/gpt.c (gpt_partition_duplicate): copy flags to new
   partition.
 * NEWS: Mention this fix.
---
 NEWS                   |    3 +++
 libparted/labels/gpt.c |   12 ++++++++++++
 2 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/NEWS b/NEWS
index b1b70b8..b180d80 100644
--- a/NEWS
+++ b/NEWS
@@ -56,6 +56,9 @@ GNU parted NEWS                                    -*- outline -*-
 
 ** Bug fixes
 
+  libparted: gpt_disk_duplicate now copies the flags over to the new
+  disk object. Previously the flags would be undefined.
+
   parted can once again create partition tables on loop devices.
   Before, "parted -s /dev/loop0 mklabel gpt" would fail.
   [bug introduced in parted-1.9.0]
diff --git a/libparted/labels/gpt.c b/libparted/labels/gpt.c
index 1f8bce1..2ef98ba 100644
--- a/libparted/labels/gpt.c
+++ b/libparted/labels/gpt.c
@@ -1366,6 +1366,18 @@ gpt_partition_duplicate (const PedPartition *part)
   result_data->type = part_data->type;
   result_data->uuid = part_data->uuid;
   strcpy (result_data->name, part_data->name);
+
+  /* Copy over all the flags */
+  result_data->lvm = part_data->lvm;
+  result_data->raid = part_data->raid;
+  result_data->boot = part_data->boot;
+  result_data->bios_grub = part_data->bios_grub;
+  result_data->hp_service = part_data->hp_service;
+  result_data->hidden = part_data->hidden;
+  result_data->msftres = part_data->msftres;
+  result_data->atvrecv = part_data->atvrecv;
+  result_data->msftrecv = part_data->msftrecv;
+  result_data->legacy_boot = part_data->legacy_boot;
   return result;
 
 error_free_part:
-- 
1.7.6.5

